src/contracts/libraries/GPv2Encoding.sol

```diff
@@ -135,68 +139,67 @@ library GPv2Encoding {
     ///     uint8 flags;
     ///     uint256 executedAmount;
     ///     uint16 feeDiscount;
-    ///     Signature {
-    ///         bytes32 r;
-    ///         bytes32 s;
-    ///         uint8 v;
-    ///     } signature;
+    ///     bytes signature;
     /// }
     /// ```
     ///
-    /// Order flags are used to encode additional order parameters such as the
-    /// kind of order, either a sell or a buy order, as well as whether the
-    /// order is partially fillable or if it is a "fill-or-kill" order. As the
-    /// most likely values are fill-or-kill sell orders, the flags are chosen
-    /// such that `0x00` represents this kind of order. The flags byte uses has
-    /// the following format:
+    /// The signature encoding depends on the scheme used to sign. The owner of
+    /// the order can be derived from the signature. See [`decodeEoaOwner`] for
+    /// encoding signatures originating from externally owned accounts (EOA).
+    ///
+    /// Trade flags are used to tightly encode information on how to decode
+    /// an order. Examples that directly affect the structure of an order are
+    /// the kind of order (either a sell or a buy order) as well as whether the
+    /// order is partially fillable or if it is a "fill-or-kill" order. It also
+    /// encodes whether the order comes from a smart contract or an EOA, in
+    /// order to choose the right signature scheme when decoding. As the most
+    /// likely values are fill-or-kill sell orders by an EOA, the flags are
+    /// chosen such that `0x00` represents this kind of order. The flags byte
+    /// uses the following format:
     /// ```
     /// bit | 7 | 6 | 5 | 4 | 3 | 2 | 1 | 0 |
     /// ----+-----------------------+---+---+
-    ///     |        unsused        | * | * |
-    ///                               |   |
-    ///                               |   +---- order kind bit, 0 for a sell
-    ///                               |         order and 1 for a buy order
-    ///                               |
-    ///                               +-------- order fill bit, 0 for fill-or-
-    ///                                         kill and 1 for a partially
-    ///                                         fillable order
+    ///     | * |      unsused      | * | * |
```

Comment:
nit: I'm not a huge fan of how the signature scheme is encoded in two placed. I would slightly prefer if two bits are used here and that the `v` value no longer be used for encoding the signature scheme such that:
```
0b00 => EIP-712 signature
0b01 => eth_sign signature
0b10 => EIP-1271 signature
0b11 => *reserved* (potentially used for delegate signature/storage approval, and if we want both, then we need another bit)
```

We could then have `decodeEIp712Signature`, `decoderEthSignSignature`, `decodeEip1271Signature`, one for each variant (with potentially a shared `decodeEcdsaSignature` for pulling `r`, `s` and `v` values from calldata).

Reply:
The above change would likely require refactoring the TS side encoding code. We could introduce a
```ts
interface Signature {
  scheme: SigningScheme,
  data: BytesLike,
}
```